src/lib/spinel/spinel.h

```diff
@@ -4951,6 +4951,60 @@ SPINEL_API_EXTERN const char *spinel_link_metrics_status_to_cstr(uint8_t status)
 
 // ----------------------------------------------------------------------------
 
+enum
+{
+    SPINEL_PROP_DISPLAY_ERROR_BUFFER_OVERFLOW       = -1,
+    SPINEL_PROP_DISPLAY_ERROR_INVALID_ARGUMENTS     = -2,
+    SPINEL_PROP_DISPLAY_ERROR_UNSUPPORTED_DATA_TYPE = -3,
+    SPINEL_PROP_DISPLAY_ERROR_INVALID_PACK_FORMAT   = -4,
+};
+
+/**
+ * This function prints a basic spinel data type (not a STRUCT or an ARRAY) to a buffer.
+ *
+ * @param[in]   data_type   The data type.
+ * @param[in]   ptr_args    If the argument in the argument list is a pointer to the value or a value.
+ * @param[in]   args        A value identifying a variable arguments list.
+ * @param[out]  buf         The output buffer.
+ * @param[in]   buf_size    The size of the output buffer.
+ *
+ * @returns If written successfully, it returns the length of bytes written. Otherwise a negative error code is
+ * returned.
+ * @retval  SPINEL_PROP_DISPLAY_ERROR_BUFFER_OVERFLOW        @p `buf_size` is not enough to write the content.
+ * @retval  SPINEL_PROP_DISPLAY_ERROR_INVALID_ARGUMENTS      @p `buf` is `NULL`.
+ * @retval  SPINEL_PROP_DISPLAY_ERROR_UNSUPPORTED_DATA_TYPE  @p `data_type` is not supported.
+ *
+ */
+SPINEL_API_EXTERN int32_t spinel_prop_display_simple_data_type(spinel_datatype_t data_type,
+                                                               bool              ptr_args,
+                                                               va_list           args,
+                                                               char *            buf,
+                                                               uint16_t          buf_size);
```

Comment:
Will different error be useful here. We can consider maing it `int` and return `-1` for all errors (which would be similar to `spinel_datatype_unpack()`.